src/ConnectedMode/ServerSentEvents/Issue/IssueServerEventsListener.cs

```diff
@@ -77,12 +80,19 @@ public async Task ListenAsync()
                         return;
                     }
 
-                    logger.LogVerbose(Resources.Suppression_IssueChangedEventReceived, issueServerEvent);
+                    var serverBranch = await branchProvider.GetServerBranchNameAsync(cancellationTokenSource.Token);
 
-                    // todo: filter for branch
-                    var issueKeys = issueServerEvent.BranchAndIssueKeys.Select(x => x.IssueKey).ToArray();
+                    var issueKeysForCurrentBranch = issueServerEvent.BranchAndIssueKeys
+                        .Where(x => x.BranchName.Equals(serverBranch))
+                        .Select(x => x.IssueKey)
+                        .ToArray();
 
-                    await suppressionIssueStoreUpdater.UpdateSuppressedIssues(issueServerEvent.IsResolved, issueKeys, cancellationTokenSource.Token);
+                    logger.LogVerbose(Resources.Suppression_IssueChangedEvent, issueServerEvent.IsResolved, string.Join(",", issueKeysForCurrentBranch));
+
+                    if (issueKeysForCurrentBranch.Any())
```

Comment:
Missing test case for "no issues for current branch"?

Reply:
It's kind of being tested in OnEvent_SingleEvent_MultipleTuples_StoreIsUpdatedWithIssuesFromCurrentBranch
--> it raises 3 separate events, one of which has no issues for current branch, and that event is ignored.